sw/device/tests/keymgr_derive_cdi_test.c

```diff
@@ -0,0 +1,448 @@
+    CHECK_ARRAYS_NE(prev_outputs->sw_key.unmasked,
+                    next_outputs->sw_key.unmasked, kKeymgrOutputSizeWords);
+    CHECK_ARRAYS_NE(prev_outputs->sideload_key, next_outputs->sideload_key,
+                    kKeymgrOutputSizeWords);
+  }
+
+  if (write) {
+    ret_sram_write_keys(next_outputs, offset);
+  } else {
+    cdi_outputs_t scratch;
+    ret_sram_read_keys(&scratch, offset);
+
+    CHECK_ARRAYS_EQ(scratch.identity.unmasked, next_outputs->identity.unmasked,
+                    kKeymgrOutputSizeWords);
+    CHECK_ARRAYS_EQ(scratch.sw_key.unmasked, next_outputs->sw_key.unmasked,
+                    kKeymgrOutputSizeWords);
+    CHECK_ARRAYS_EQ(scratch.sideload_key, next_outputs->sideload_key,
+                    kKeymgrOutputSizeWords);
+  }
+}
+
+/**
+ * This test implements both the `chip_sw_keymgr_derive_sealing` and
+ * `chip_sw_keymgr_derive_attestation` testplan items, i.e.,
+ * verifies the validity of the sealing and attestation CDI flows. The
+ * individual testpoint requirements are as follows:
+ *
+ * - For each keymgr operational state: `CreatorRootKey`, `OwnerIntKey` and
+ *   `OwnerKey`:
+ *   - Generate identity SW output for the Attestation CDI.
+ *   - Generate SW output for the Attestation CDI.
+ *   - Generate OTBN sideload output for the Attestation CDI.
+ *
+ * The creation of the three outputs is handled by the `derive_identity`,
+ * `derive_sw_key` and `derive_sideload_otbn_key` functions, which are invoked
+ * by `derive_keys` through `test_derive_cdi` that traverses the three
+ * operational states.
+ *
+ * - Ensure that the key output changes after calculating the previous steps
+ *   after a keymgr advance operation.
+ *
+ * The generated CDI outputs in state `i` are passed to state `i+1` and checked
+ * in `derive_keys` that they differ.
+ *
+ * - The keymgr shall be able to reproduce the same keys for a give device
+ *   configuration and known set of inputs.
+ *
+ * The CDI outputs in each state are written into the retention SRAM. After
+ * having cycled through all the states, a software reset is performed and
+ * CDI output generation is repeated making sure that the same keys are being
+ * created (same as the keys in the retention SRAM from the first run) as the
+ * device configuration and inputs are unchanged. Reading/writing and key
+ * comparison all happen in `derive_keys`.
+ *
+ * - The software binding registers must be locked after configuration until a
+ *   keymgr advance operation.
+ *
+ * This check is implemented in the keymgr DIF (see `dif_keymgr_advance_state`).
+ *
+ * - The device initial state needs to be equivalent to the end state of the
+ *   `manuf_ft_provision_rma_token_and_personalization` testpoint, otherwise
+ *   the keymgr will fail to advance into operational states.
+ *
+ * The provisioning of the device secrets in the OTP is handled by the keymgr
+ * testutils (see `keymgr_testutils_init_nvm_then_reset`).
```

Comment:
I would just reword the test point to say:
```suggestion
 * - OTP SECRET2 partition (which contains the CREATOR_ROOT_KEY) and flash info  
 *   pages 1 and 2 in partition 0 must be configured, otherwise the keymgr will fail to
 *   advance into operational states.
 *
 * The provisioning of the device secrets in the OTP SECRET2 partition is handled by the
 * Bazel execution environment, and the flash info secrets are handled by the keymgr
 * testutils (see `keymgr_testutils_init_nvm_then_reset`).
```
I think the test point description was a bit outdated/less concise than it should be, but this is what your requirement is (and you are correct in that the test already ensures this as described).